Common/FiguresStorage/src/FiguresStorage.cpp

```diff
@@ -15,227 +17,155 @@ FiguresStorage::~FiguresStorage()
     }
 }
 
-namespace details
+void addShapeParams(Shape shape, const json& data)
```

Comment:
Ну ты теперь копию делаешь. Я же просил по ссылке
```suggestion
void addShapeParams(Shape& shape, const json& data)
```
Заполняешь копию теперь